packages/firestore/src/local/indexeddb_remote_document_cache.ts

```diff
@@ -217,6 +220,40 @@ export class IndexedDbRemoteDocumentCache implements RemoteDocumentCache {
     const range = IDBKeyRange.upperBound(changeId);
     return documentChangesStore(transaction).delete(range);
   }
+
+  resetLastProcessedDocumentChange(
+    transaction: PersistenceTransaction
+  ): PersistencePromise<void> {
+    const store = documentChangesStore(transaction);
+    return this.synchronizeLastDocumentChangeId(store);
+  }
+
+  private synchronizeLastDocumentChangeId(
+    documentChangesStore: SimpleDbStore<
+      DbRemoteDocumentChangesKey,
+      DbRemoteDocumentChanges
+    >
+  ): PersistencePromise<void> {
+    // If there are no existing changes, we set `lastProcessedDocumentChangeId`
+    // to 0 since IndexedDb's auto-generated keys start at 1.
+    this._lastProcessedDocumentChangeId = 0;
+    return documentChangesStore.iterate(
+      { keysOnly: true, reverse: true },
+      (key, value, control) => {
+        this._lastProcessedDocumentChangeId = key;
+        control.done();
+      }
+    );
+  }
+}
+
+export function isRemoteDocumentChangesGarbageCollectedError(
```

Comment:
I like verbosity, but "is...Error" gets hard to parse when there are 5 words in between.  How about isDocumentChangeMissingError()?

I was also wondering if you could just:
```
export const DOCUMENT_CHANGE_MISSING_ERROR = new Error(...);
```

But I am not sure how JS works. Would that mess up the stack trace in the error or something?

It's also a little dubious to be using errors for control flow.  We could change getNewDocumentChanges() to return null if it encountered a missing change or something, but I think I'm not that thrilled about the idea.

Reply:
I changed the name to `isDocumentChangeMissingError`.

In general, I would like to shy away from statically allocating errors as it does mess up the stacktraces (even in JS).

I'm gonna use the current fire alarm in SFO as an excuse to not comment on the API contract. I can't really think of a much better way to convey this information.